Replace cargo2nix with Crane and optimize Flake

lafrenierejm opened the pull request:
Crane obsoletes the need for `Cargo.nix` by parsing dependencies directly from `Cargo.lock` and improves build times by building and caching dependencies as their own Nix derivation.

teto commented on .envrc:
```diff
@@ -1 +1,3 @@
-use nix
+if command -v nix &>/dev/null; then
+    use flake
+fi
```
I would think direnv already does the check when setting `use flake` ?

lafrenierejm replied:
When I last tested it multiple months ago, direnv would fail if it didn't have the flake module loaded and I omitted the `command` check. It's definitely possible that has changed in recent versions of direnv, though; feel free to send a link to a changelog entry or commit and I'll gladly simplify this.

teto commented on flake.nix:
```diff
@@ -1,30 +1,123 @@
-          
-          default = mujmap;
+
+        pre-commit = pre-commit {
```
maybe submit this in another PR ? I dont really like this pre-commit stuff. How can I disable it by default ? there is an upcoming official nix formatter, I would prefer using that to alejandra.

lafrenierejm replied:
> maybe submit this in another PR ?

Sure, I can remove the pre-commit hooks from this PR.

lafrenierejm replied:
Removed the pre-commit hooks in e30a8310221ada3b0a376c2fa7de5e9363646063.

teto commented on flake.nix:
```diff
@@ -1,30 +1,123 @@
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    advisory-db = {
+      url = "github:rustsec/advisory-db";
```
is that to check for vunerabilities ?

lafrenierejm replied:
Yes, it provides a way to audit dependencies for vulnerabilities via https://rustsec.org/.

teto commented:
I dont know what is the best now for nix + rust. I think I used crane for another package. Per package caching sounds nice, and the removal of Cargo.nix is awesome too. I think I approve but I am not maintainer on the repo.

teto commented:
@lafrenierejm would you like to become a maintainer ? maybe @elizagamedev could give you access if positive ?

lafrenierejm committed:
Replace cargo2nix with Crane and optimize Flake

Crane obsoletes the need for `Cargo.nix` by parsing dependencies directly from
`Cargo.lock` and improves build times by building and caching dependencies as
their own Nix derivation.

lafrenierejm commented:
> would you like to become a maintainer ?

@teto I would be glad to help maintain this project. The call is entirely @elizagamedev's, though.
